src/System.Private.CoreLib/shared/System/Threading/Semaphore.Windows.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    public sealed partial class Semaphore
+    {
+        private const uint AccessRights = (uint)Interop.Kernel32.MAXIMUM_ALLOWED | Interop.Kernel32.SYNCHRONIZE | Interop.Kernel32.SEMAPHORE_MODIFY_STATE;
+
+        private Semaphore(SafeWaitHandle handle)
+        {
+            SafeWaitHandle = handle;
+        }
+
+        private void CreateSemaphoreCore(int initialCount, int maximumCount, string name, out bool createdNew)
+        {
+            Debug.Assert(initialCount >= 0);
+            Debug.Assert(maximumCount >= 1);
+            Debug.Assert(initialCount <= maximumCount);
+
+#if PLATFORM_UNIX
```

Comment:
Can we make this instead `#if !PLATFORM_WINDOWS`?  I just got rid of the last uses of PLATFORM_UNIX in corelib:
https://github.com/dotnet/coreclr/pull/18411